addon/components/o-s-s/chip.ts

```diff
@@ -0,0 +1,66 @@
+import Component from '@glimmer/component';
+import { assert } from '@ember/debug';
+import { action } from '@ember/object';
+
+export type SkinType = 'default' | 'primary' | 'success' | 'danger';
+
+type SkinDefType = {
+  [key in SkinType]: string;
+};
+
+export const SkinDefinition: SkinDefType = {
+  default: 'default',
+  primary: 'primary',
+  success: 'success',
+  danger: 'danger'
+};
+
+const BASE_CLASS_CHIP: string = 'upf-chip';
+interface OSSChipArgs {
+  skin?: string;
+  label?: string;
+  onClick(crossClick: boolean): void;
+  disabled?: boolean;
+  ellipsisWidth?: number;
+}
+
+export default class OSSChip extends Component<OSSChipArgs> {
+  constructor(owner: unknown, args: OSSChipArgs) {
+    super(owner, args);
+
+    assert(
+      '[component][OSS::Chip] The parameter @onClick of type function is mandatory',
+      typeof this.args.onClick === 'function'
+    );
+  }
+
+  get skin(): string {
+    return SkinDefinition[this.args.skin as SkinType] ?? SkinDefinition.primary;
+  }
+
+  get ellipsisWidth(): string {
+    return this.args.ellipsisWidth ? `text-ellipsis-${this.args.ellipsisWidth}` : '';
+  }
```

Comment:
My opinion, we'll see what the others say but :
I would not do this with the oss classes `text-ellipsis-xxx`. There are too few sizes defined, there is no quick way for a dev to know them when using the component (except from opening storybook or the oss project) and they are kind of strict.

I would use the size provided in the component params to force a max-size to the chip ; and then handle the ellipsis if needed when the content overflows.

Reply:
Its really more about matching the designs, not what we think :)
I'll bet that Victor would set max-widths to something rounded.

Also we've seen the overwhelming power of the flex-ellipsis over the text-ellipsis in the past :trollface: